mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp

```diff
@@ -525,20 +557,53 @@ class MatMulConverter : public OpConversionPattern<tosa::MatMulOp> {
       dynDims[2] = rewriter.create<tensor::DimOp>(loc, op->getOperand(1), 2);
     }
 
+    auto getTypeWithoutBatch = [&](ShapedType ty) {
+      auto shape2D = {ty.getDimSize(1), ty.getDimSize(2)};
+      return RankedTensorType::get(shape2D, ty.getElementType());
+    };
+
     SmallVector<Value> filteredDims = condenseValues(dynDims);
 
+    bool isBatchOneDim =
+        outputTy.getDimSize(0) == 1 && this->useMatmulForSingleBatch;
```

Comment:
nit: useMatmulForSingleBatch doesn't really relate to whether `isBatchOneDim`

Reply:
I'll rename it to a local `useMatmul`.

Reply:
Done!